lib/puppet/graph/simple_graph.rb

```diff
@@ -4,6 +4,8 @@
 
 # A hopefully-faster graph class to replace the use of GRATR.
 class Puppet::Graph::SimpleGraph
+  include Puppet::Util::PsychSupport
```

Comment:
Can you break this out into a separate PR for PUP-7381, since that one is about using `encode_with` consistently, and this PR is about removing duplication?